### ISSUE TITLE
BUG: Formats JSON when response type isn't JSON and body contains more

### ISSUE THREAD

**Issue by rudiedirkx** [callumlocke/json-formatter#57]
I'm testing my API via web and I have a `text/plain` response where the body is (literally, no headers):

```
GET /api/user HTTP/1.1
User-Agent: Super ultra fast super HTTP browser
Host: xxxxxx.xx
Accept: */*



HTTP/1.1 200 OK
Date: Sat, 25 Apr 2015 17:52:51 GMT
Content-Length: 62
Content-Type: application/json; charset=utf-8

{"error":"Access denied","zone":"LOGGED_IN","uri":"api\/user"}
````

And somehow it's formatted into:

```
{
  "error": "Access denied",
  "zone": "LOGGED_IN",
  "uri": "api/user"
}
```

Indeed, the response contains **some** what-looks-like-JSON, but don't just assume and discard the rest!

Just to extra clarify: the first big paste with request header, response header and response body IS the page. They're not the request and response from browser.

**Comment by callumlocke** [comment]
Can you provide the URL?

**Comment by callumlocke** [comment]
Please show me a URL if you want me to diagnose.

This extension doesn't look at mime types, it just gets invoked on any page that Chrome would normally display as plain text (ie, everything except HTML pages and images). If the entire plain text response body parses as JSON, it formats it. It shouldn't discard anything.

Are you saing those what-look-like HTTP headers are actually supposed to be in the response body, and the extension is discarding them ? If so that sounds sounds unlikely. The response body wouldn't parse as JSON.

If you can't provide a public URL for me to look at, try disabling the extension and then load the page, and tell me if the response body really contains those headers.

**Comment by rudiedirkx** [comment]
> Are you saing those what-look-like HTTP headers are actually supposed to be in the response body, and the extension is discarding them?

Yes, exactly that. Somehow it parses only the JSON part and not the rest. Very strange.

When I disable the extension (or click RAW) it displays all of that: request headers, response headers, response body. Like I said, I was debugging my API, so printing a lot of API requests etc.

I don't have a public URL, but I can make a quick video, I think.

**Comment by rudiedirkx** [comment]
Voila: http://screencast.com/t/iBQUF9zM

It starts and ends with the extension disabled, so Chrome displays plain text data plainly in plain text format.

**Comment by callumlocke** [comment]
wow that is weird... Can you try something else: disable the extension, load the page, right click anywhere on the raw text and Inspect Element, and show me how it looks in the Elements tab? (With the extension disabled - it should appear in a PRE element, I just want to see if the whole body really is in the PRE)

Thanks

**Comment by rudiedirkx** [comment]
See for yourself: https://gist.githubusercontent.com/rudiedirkx/4aae7ee433c5511f8415/raw/cd412ea24cdd6943e62097adc4bad9d39122eac5/http-req-res.txt

**Comment by callumlocke** [comment]
There's always a DOM. Chrome creates a makeshift DOM for plain text files (usually just a body with a PRE element). I need to see that to understand what's going on. With the extension disabled.

**Comment by rudiedirkx** [comment]
The link doesn't 'work' for you? Weird.

When I disable the extension and load the plain text gist, this is the DOM:

```
<html>
<head>
<style type="text/css"></style>
</head>
<body>
<pre style="word-wrap: break-word; white-space: pre-wrap;">GET /api/user HTTP/1.1
User-Agent: Super ultra fast super HTTP browser
Host: xxxxxx.xx
Accept: */*



HTTP/1.1 200 OK
Date: Sat, 25 Apr 2015 17:52:51 GMT
Content-Length: 62
Content-Type: application/json; charset=utf-8

{"error":"Access denied","zone":"LOGGED_IN","uri":"api\/user"}
</pre>
</body>
</html>
```

With the extension enabled the pre still exists, but with `[hidden]`. I guess you know that. And it adds `#optionBar` and  `#jfContent` in `body`

**Comment by rudiedirkx** [comment]
I'm wildly guessing this is the culprit:

```
// Strip any leading garbage, such as a 'while(1);'
var strippedText = text.substring( firstJSONCharIndex(text) ) ;
```

IMO you shouldn't strip anything and only format JSON content, but it seems that was an intentional choice. Up to you now.

**Comment by callumlocke** [comment]
Ah I get you now. The stripping thing was intended to get rid of stuff that some public APIs put before JSONP content to prevent cross-site stuff. It's obviously too crude. Sorry for taking a while to get this.

I'm working on a new version that takes a completely different approach, so this won't happen any more.

**Comment by BastianHofmann** [comment]
Care to push the WIP version? Maybe we can help out.

**Comment by ryanpcmcquen** [comment]
Any update on this? I love this plugin, but this bug sometimes hurts my soul ...
